store/src/genesis.rs

```diff
@@ -1,21 +1,99 @@
-use miden_crypto::merkle::{EmptySubtreeRoots, MmrPeaks, TieredSmt};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use miden_crypto::{
+    dsa::rpo_falcon512,
+    merkle::{EmptySubtreeRoots, MerkleError, MmrPeaks, SimpleSmt, TieredSmt},
+    Felt,
+};
+use miden_lib::{faucets::create_basic_fungible_faucet, wallets::create_basic_wallet, AuthScheme};
 use miden_node_proto::block_header;
-use miden_objects::{notes::NOTE_LEAF_DEPTH, Digest};
+use miden_objects::{
+    accounts::Account, assets::TokenSymbol, notes::NOTE_LEAF_DEPTH, BlockHeader, Digest,
+};
+use serde::{Deserialize, Serialize};
 
 use crate::state::ACCOUNT_DB_DEPTH;
 
-/// Generates the header of the genesis block. The timestamp is currently hardcoded to be the UNIX epoch.
-pub fn genesis_header() -> block_header::BlockHeader {
-    block_header::BlockHeader {
-        prev_hash: Some(Digest::default().into()),
-        block_num: 0,
-        chain_root: Some(MmrPeaks::new(0, Vec::new()).unwrap().hash_peaks().into()),
-        account_root: Some(EmptySubtreeRoots::entry(ACCOUNT_DB_DEPTH, 0).into()),
-        nullifier_root: Some(TieredSmt::default().root().into()),
-        note_root: Some(EmptySubtreeRoots::entry(NOTE_LEAF_DEPTH, 0).into()),
-        batch_root: Some(Digest::default().into()),
-        proof_hash: Some(Digest::default().into()),
-        version: 0,
-        timestamp: 0,
+/// Represents the state at genesis, which will be used to derive the genesis block.
+#[derive(Serialize, Deserialize)]
+pub struct GenesisState {
+    pub accounts: Vec<Account>,
+    pub version: u64,
+    pub timestamp: u64,
+}
+
+impl GenesisState {
+    pub fn new(pub_key: rpo_falcon512::PublicKey) -> Self {
+        let accounts = {
+            let mut accounts = Vec::new();
+
+            // fungible asset faucet
+            {
+                let (account, _) = create_basic_fungible_faucet(
+                    [0; 32],
+                    TokenSymbol::new("TODO").unwrap(),
+                    9,
+                    Felt::from(1_000_000_000_u64),
+                    AuthScheme::RpoFalcon512 { pub_key },
+                )
+                .unwrap();
+
+                accounts.push(account);
+            }
+
+            // basic wallet account
+            {
+                let (account, _) = create_basic_wallet(
+                    [0; 32],
+                    AuthScheme::RpoFalcon512 { pub_key },
+                    miden_objects::accounts::AccountType::RegularAccountUpdatableCode,
+                )
+                .unwrap();
+
+                accounts.push(account);
+            }
+
+            accounts
+        };
```

Comment:
In the future, we may specify a more flexible way to do this, but I think for now it is to hard-code some of these values. A couple of comments/questions though:

- We probably shouldn't use the same value for the seeds (I don't think it'll break anything, but still). So, maybe for fungible faucet we use all zeros and for the account we use all ones?
- Should we use `POL` for the test token symbol?
- We should probably use two different public keys for these accounts. The workflow here could be as follows:
  - The genesis utility generates two Falcon key-pairs using similar seeds as we used for accounts (i.e., all zeros and all ones). This can be done via `KeyPair::from_seed()` method.
  - The accounts are created using the public keys from these key pairs.
  - Private keys are output along size the genesis JSON file. For this we'd need to serialize `KeyPair` struct. For now, we can probably just use `KeyPair::to_bytes()` and name files something like `faucet.fsk` and `account.fsk`. In the future, we can investigate more standard ways to encode key material.